Selenium.Axe/Selenium.Axe.Test/IntegrationTests.cs

```diff
@@ -70,6 +71,84 @@ public void RunScanOnGivenElement(string browser)
             results.Violations.Should().HaveCount(3);
         }
 
+        [TestMethod]
+        [DataRow("Chrome")]
+        [DataRow("Firefox")]
+        public void ReportFullPage(string browser)
+        {
+            string path = CreateReportPath();
+            this.InitDriver(browser);
+            LoadTestPage();
+
+            var mainElement = _wait.Until(drv => drv.FindElement(By.TagName("main")));
+            _webDriver.CreateReport(path);
+
+            ValidateReport(path, 4, 23);
+        }
+
+        [TestMethod]
+        [DataRow("Chrome")]
+        [DataRow("Firefox")]
+        public void ReportOffElement(string browser)
+        {
+            string path = CreateReportPath();
+            this.InitDriver(browser);
+            LoadTestPage();
+
+            var mainElement = _wait.Until(drv => drv.FindElement(By.CssSelector("main")));
+            _webDriver.CreateReport(mainElement, path);
+
+            ValidateReport(path, 3, 15);
+        }
+
+        [TestMethod]
+        [DataRow("Chrome")]
+        [DataRow("Firefox")]
+        public void ReportRespectRules(string browser)
+        {
+            string path = CreateReportPath();
+            this.InitDriver(browser);
+            LoadTestPage();
+            var mainElement = _wait.Until(drv => drv.FindElement(By.CssSelector("main")));
+
+            var builder = new AxeBuilder(_webDriver).DisableRules("color-contrast");
+            _webDriver.CreateReport(builder.Analyze(), path);
+
+            ValidateReport(path, 3, 18);
+        }
+
+        private string CreateReportPath()
+        {
+            string codeBase = System.Reflection.Assembly.GetExecutingAssembly().CodeBase;
+            UriBuilder uri = new UriBuilder(codeBase);
+            string path = Uri.UnescapeDataString(uri.Path);
+            return Path.Combine(Path.GetDirectoryName(path), Guid.NewGuid() + ".html");
```

Comment:
Consider adding a test case that analyzes a produced report file for axe-core violations?

Reply:
Added validation against a raw Axe json results.
Note- helped identify a couple of bugs I missed :), thanks!